test/integration/template_insights_test.go

```diff
@@ -0,0 +1,240 @@
+package integration
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+
+	"github.com/nuodb/nuodb-helm-charts/test/testlib"
+)
+
+func checkSidecarContainers(t *testing.T, containers []v1.Container, options *helm.Options, chartPath string) {
+	assert.NotEmpty(t, containers)
```

Comment:
I am assuming that you want to abandon the test straight away when this assert is false. In stretch, that's actually `require.NotEmpty`, not `assert.`

Reply:
Not sure if we have a convention when using `assert` or `require`. I normally use `assert` in the actual test and `require` in `testlib` utilities.

Reply:
it is not about convention. `assert` marks the test as failed, but continues. So in this case, it will let marked as failed, but further operations can still be executed. To give you an example:

container := nil
assert.NotNil(t, container)
assert.Greater(t, size(container), 2) // we assume that there are two elements
if container[0]... <- this will panic

Reply:
I've updated the line as per your recommendation.